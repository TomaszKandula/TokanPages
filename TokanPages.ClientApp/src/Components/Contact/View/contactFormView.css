.contact-card-no-shadow {
    box-shadow: none;
}
