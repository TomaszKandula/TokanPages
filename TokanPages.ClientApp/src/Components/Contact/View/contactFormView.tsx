import * as React from "react";
import Container from "@material-ui/core/Container";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Button from "@material-ui/core/Button";
import { CircularProgress } from "@material-ui/core";
import Skeleton from "@material-ui/lab/Skeleton";
import { VioletCheckbox } from "../../../Theme";
import { ViewProperties } from "../../../Shared/Abstractions";
import { ReactChangeEvent, ReactKeyboardEvent } from "../../../Shared/types";
import { ContactFormStyle } from "./contactFormStyle";

interface Properties extends ViewProperties {
    caption: string;
    text: string;
    keyHandler: (event: ReactKeyboardEvent) => void;
    formHandler: (event: ReactChangeEvent) => void;
    firstName: string;
    lastName: string;
    email: string;
    subject: string;
    message: string;
    terms?: boolean;
    buttonHandler: () => void;
    progress: boolean;
    buttonText: string;
    consent: string;
    labelFirstName: string;
    labelLastName: string;
    labelEmail: string;
    labelSubject: string;
    labelMessage: string;
    multiline?: boolean;
    minRows?: number;
}

const ActiveButton = (props: Properties): JSX.Element => {
    const classes = ContactFormStyle();
    return (
        <Button
            fullWidth
            type="submit"
            variant="contained"
            onClick={props.buttonHandler}
            disabled={props.progress}
            className={classes.button}
        >
            {!props.progress ? props.buttonText : <CircularProgress size={20} />}
        </Button>
    );
};

export const ContactFormView = (props: Properties): JSX.Element => {
    const classes = ContactFormStyle();
    return (
        <section className={classes.section}>
            <Container maxWidth="lg">
                <Container maxWidth="sm">
                    <Box pt={8} pb={10}>
                        <Box mb={6} textAlign="center" data-aos="fade-down">
                            <Typography gutterBottom={true} className={classes.caption}>
                                {props.isLoading ? <Skeleton variant="text" /> : props.caption?.toUpperCase()}
                            </Typography>
                        </Box>
                        <Box>
                            <Grid container spacing={2}>
                                <Grid item xs={12} sm={6}>
                                    <div data-aos="zoom-in">
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="100%" height="45px" />
                                        ) : (
                                            <TextField
                                                required
                                                fullWidth
                                                id="firstName"
                                                name="firstName"
                                                autoComplete="fname"
                                                variant="outlined"
                                                onKeyUp={props.keyHandler}
                                                onChange={props.formHandler}
                                                value={props.firstName}
                                                label={props.labelFirstName}
                                            />
                                        )}
                                    </div>
                                </Grid>
                                <Grid item xs={12} sm={6}>
                                    <div data-aos="zoom-in">
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="100%" height="45px" />
                                        ) : (
                                            <TextField
                                                required
                                                fullWidth
                                                id="lastName"
                                                name="lastName"
                                                autoComplete="lname"
                                                variant="outlined"
                                                onKeyUp={props.keyHandler}
                                                onChange={props.formHandler}
                                                value={props.lastName}
                                                label={props.labelLastName}
                                            />
                                        )}
                                    </div>
                                </Grid>
                                <Grid item xs={12}>
                                    <div data-aos="zoom-in">
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="100%" height="45px" />
                                        ) : (
                                            <TextField
                                                required
                                                fullWidth
                                                id="email"
                                                name="email"
                                                autoComplete="email"
                                                variant="outlined"
                                                onKeyUp={props.keyHandler}
                                                onChange={props.formHandler}
                                                value={props.email}
                                                label={props.labelEmail}
                                            />
                                        )}
                                    </div>
                                </Grid>
                                <Grid item xs={12}>
                                    <div data-aos="zoom-in">
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="100%" height="45px" />
                                        ) : (
                                            <TextField
                                                required
                                                fullWidth
                                                id="subject"
                                                name="subject"
                                                autoComplete="subject"
                                                variant="outlined"
                                                onKeyUp={props.keyHandler}
                                                onChange={props.formHandler}
                                                value={props.subject}
                                                label={props.labelSubject}
                                            />
                                        )}
                                    </div>
                                </Grid>
                                <Grid item xs={12}>
                                    <div data-aos="zoom-in">
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="100%" height="45px" />
                                        ) : (
                                            <TextField
                                                required
                                                fullWidth
                                                multiline={props.multiline}
                                                minRows={props.minRows}
                                                id="message"
                                                name="message"
                                                autoComplete="message"
                                                variant="outlined"
                                                onChange={props.formHandler}
                                                value={props.message}
                                                label={props.labelMessage}
                                            />
                                        )}
                                    </div>
                                </Grid>
                                <Grid item xs={12}>
                                    <div data-aos="zoom-in">
                                        {props.isLoading ? (
                                            <Skeleton variant="rect" width="100%" height="30px" />
                                        ) : (
                                            <FormControlLabel
                                                control={
                                                    <VioletCheckbox
                                                        onChange={props.formHandler}
                                                        checked={props.terms}
                                                        name="terms"
                                                        id="terms"
                                                    />
                                                }
                                                label={props.consent}
                                            />
                                        )}
                                    </div>
                                </Grid>
                            </Grid>
                            <Box my={2} data-aos="fade-up">
                                {props.isLoading ? (
                                    <Skeleton variant="rect" width="100%" height="40px" />
                                ) : (
                                    <ActiveButton {...props} />
                                )}
                            </Box>
                        </Box>
                    </Box>
                </Container>
            </Container>
        </section>
    );
};
