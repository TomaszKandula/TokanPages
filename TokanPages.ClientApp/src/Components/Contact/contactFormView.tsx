import * as React from "react";
import Container from "@material-ui/core/Container";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Button from "@material-ui/core/Button";
import { CircularProgress } from "@material-ui/core";
import Skeleton from "@material-ui/lab/Skeleton";
import contactFormStyle from "./contactFormStyle";
import VioletCheckbox from "../../Theme/customCheckboxes";

interface IBinding 
{
    bind: IProperties;
}

interface IProperties
{
    isLoading: boolean;
    caption: string;
    text: string;
    formHandler: any;
    firstName: string;
    lastName: string;
    email: string;
    subject: string;
    message: string;
    terms?: boolean;
    buttonHandler: any;
    progress: boolean;
    buttonText: string;
    consent: string;
}

const ContactFormView = (props: IBinding): JSX.Element =>
{
    const classes = contactFormStyle();
    return (
        <section className={classes.section}>
            <Container maxWidth="lg">
                <Container maxWidth="sm">
                    <div data-aos="fade-up">
                        <Box pt={8} pb={10}>
                            <Box mb={6} textAlign="center">
                                <Typography gutterBottom={true} className={classes.caption}>
                                    {props.bind?.isLoading ? <Skeleton variant="text" /> : props.bind?.caption?.toUpperCase()}
                                </Typography>
                            </Box>
                            <Box>
                                <Grid container spacing={2}>
                                    <Grid item xs={12} sm={6}>
                                        <TextField 
                                            onChange={props.bind?.formHandler} value={props.bind?.firstName} variant="outlined" 
                                            required fullWidth name="firstName" id="firstName" label="First name" autoComplete="fname" 
                                        />
                                    </Grid>
                                    <Grid item xs={12} sm={6}>
                                        <TextField 
                                            onChange={props.bind?.formHandler} value={props.bind?.lastName} variant="outlined" 
                                            required fullWidth name="lastName" id="lastName" label="Last name" autoComplete="lname" 
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <TextField 
                                            onChange={props.bind?.formHandler} value={props.bind?.email} variant="outlined" 
                                            required fullWidth name="email" id="email" label="Email address" autoComplete="email" 
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <TextField 
                                            onChange={props.bind?.formHandler} value={props.bind?.subject} variant="outlined" 
                                            required fullWidth name="subject" id="subject" label="Subject" autoComplete="subject" 
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <TextField 
                                            onChange={props.bind?.formHandler} value={props.bind?.message} variant="outlined" 
                                            required multiline minRows={6} fullWidth autoComplete="message" name="message" id="message" label="Message" 
                                        />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <FormControlLabel 
                                            control={<VioletCheckbox onChange={props.bind?.formHandler} checked={props.bind?.terms} name="terms" id="terms" />} 
                                            label={props.bind?.consent} 
                                        />
                                    </Grid>
                                </Grid>
                                <Box my={2}>
                                    <Button fullWidth onClick={props.bind?.buttonHandler} type="submit" variant="contained" disabled={props.bind?.progress} className={classes.button}>
                                        {props.bind?.progress &&  <CircularProgress size={20} />}
                                        {!props.bind?.progress && props.bind?.buttonText}
                                    </Button>
                                </Box>
                            </Box>
                        </Box>
                    </div>
                </Container>
            </Container>
        </section>
    );
}

export default ContactFormView;
