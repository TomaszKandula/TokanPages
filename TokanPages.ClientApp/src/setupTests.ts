// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
