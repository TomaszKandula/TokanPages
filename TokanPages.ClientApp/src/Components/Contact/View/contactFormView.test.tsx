import "../../../setupTests";
import React from "react";
import { render } from "enzyme";
import { ContactFormView } from "./contactFormView";

describe("test component: contactFormView", () => {
    it("should render correctly '<ContactFormView />' when content is loaded.", () => {
        const html = render(
            <ContactFormView
                isLoading={false}
                caption="Contact me"
                text="If you have any questions..."
                hasBackButton={true}
                keyHandler={jest.fn()}
                formHandler={jest.fn()}
                firstName="Ester"
                lastName="Exposito"
                email="ester.exposito@gmail.com"
                subject="Test subject"
                message="Test message..."
                terms={false}
                buttonHandler={jest.fn()}
                progress={false}
                buttonText="Submit"
                consent="I agree!"
                labelFirstName="First name"
                labelLastName="Last name"
                labelEmail="Email address"
                labelSubject="Subject"
                labelMessage="Message"
                multiline={false}
                minRows={undefined}
            />
        );

        expect(html).toMatchSnapshot();
    });
});
