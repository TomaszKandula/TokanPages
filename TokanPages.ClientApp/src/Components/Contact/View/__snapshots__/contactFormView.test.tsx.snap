// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: contactFormView should render correctly '<ContactFormView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="bulma-container bulma-is-max-desktop"
        >
          <div
            class="p-6"
          >
            <div
              data-aos="fade-down"
              data-aos-delay="0"
            >
              <p
                class="is-size-3	has-text-centered has-text-link"
              >
                CONTACT ME
              </p>
            </div>
            <div
              class="px-6"
            >
              <div
                class="bulma-card "
              >
                <div
                  class="bulma-card-content background-colour-inherited"
                >
                  <div
                    class="is-flex is-flex-direction-column is-align-items-center"
                  >
                    <div>
                      <svg
                        class="is-flex is-align-self-center has-text-link"
                        role="presentation"
                        style="width: 4.5rem; height: 4.5rem;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M21,8V7L18,9L15,7V8L18,10M22,3H2A2,2 0 0,0 0,5V19A2,2 0 0,0 2,21H22A2,2 0 0,0 24,19V5A2,2 0 0,0 22,3M8,6A3,3 0 0,1 11,9A3,3 0 0,1 8,12A3,3 0 0,1 5,9A3,3 0 0,1 8,6M14,18H2V17C2,15 6,13.9 8,13.9C10,13.9 14,15 14,17M22,12H14V6H22"
                          style="fill: currentColor;"
                        />
                      </svg>
                    </div>
                    <p
                      class="is-size-3 has-text-grey"
                    >
                      Contact me
                    </p>
                  </div>
                  <div
                    class="pt-5"
                  >
                    <div
                      class="bulma-columns"
                    >
                      <div
                        class="bulma-column"
                      >
                        <div
                          data-aos="zoom-in"
                          data-aos-delay="0"
                        >
                          <div
                            class="bulma-control  "
                          >
                            <input
                              autocomplete="fname"
                              class="bulma-input   "
                              id="firstName"
                              name="firstName"
                              placeholder="First name"
                              required=""
                              type="text"
                              value="Ester"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="bulma-column"
                      >
                        <div
                          data-aos="zoom-in"
                          data-aos-delay="0"
                        >
                          <div
                            class="bulma-control  "
                          >
                            <input
                              autocomplete="lname"
                              class="bulma-input   "
                              id="lastName"
                              name="lastName"
                              placeholder="Last name"
                              required=""
                              type="text"
                              value="Exposito"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="bulma-columns"
                    >
                      <div
                        class="bulma-column"
                      >
                        <div
                          data-aos="zoom-in"
                          data-aos-delay="0"
                        >
                          <div
                            class="bulma-control  "
                          >
                            <input
                              autocomplete="email"
                              class="bulma-input   "
                              id="email"
                              name="email"
                              placeholder="Email address"
                              required=""
                              type="text"
                              value="ester.exposito@gmail.com"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="bulma-columns"
                    >
                      <div
                        class="bulma-column"
                      >
                        <div
                          data-aos="zoom-in"
                          data-aos-delay="0"
                        >
                          <div
                            class="bulma-control  "
                          >
                            <input
                              autocomplete="subject"
                              class="bulma-input   "
                              id="subject"
                              name="subject"
                              placeholder="Subject"
                              required=""
                              type="text"
                              value="Test subject"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="bulma-columns"
                    >
                      <div
                        class="bulma-column"
                      >
                        <div
                          data-aos="zoom-in"
                          data-aos-delay="0"
                        >
                          <div
                            class="bulma-field"
                          >
                            <div
                              class="bulma-control  "
                            >
                              <textarea
                                class="bulma-textarea   bulma-has-fixed-size"
                                id="message"
                                name="message"
                                placeholder="Message"
                                required=""
                              >
                                Test message...
                              </textarea>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="bulma-content"
                    >
                      <div
                        class="bulma-notification"
                      >
                        <p
                          class="is-size-6"
                        >
                          I agree!
                        </p>
                      </div>
                      <div
                        data-aos="fade-up"
                        data-aos-delay="0"
                      >
                        <button
                          class="bulma-button bulma-is-link bulma-is-light bulma-is-fullwidth"
                          type="submit"
                        >
                          Submit
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="bulma-container bulma-is-max-desktop"
      >
        <div
          class="p-6"
        >
          <div
            data-aos="fade-down"
            data-aos-delay="0"
          >
            <p
              class="is-size-3	has-text-centered has-text-link"
            >
              CONTACT ME
            </p>
          </div>
          <div
            class="px-6"
          >
            <div
              class="bulma-card "
            >
              <div
                class="bulma-card-content background-colour-inherited"
              >
                <div
                  class="is-flex is-flex-direction-column is-align-items-center"
                >
                  <div>
                    <svg
                      class="is-flex is-align-self-center has-text-link"
                      role="presentation"
                      style="width: 4.5rem; height: 4.5rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M21,8V7L18,9L15,7V8L18,10M22,3H2A2,2 0 0,0 0,5V19A2,2 0 0,0 2,21H22A2,2 0 0,0 24,19V5A2,2 0 0,0 22,3M8,6A3,3 0 0,1 11,9A3,3 0 0,1 8,12A3,3 0 0,1 5,9A3,3 0 0,1 8,6M14,18H2V17C2,15 6,13.9 8,13.9C10,13.9 14,15 14,17M22,12H14V6H22"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </div>
                  <p
                    class="is-size-3 has-text-grey"
                  >
                    Contact me
                  </p>
                </div>
                <div
                  class="pt-5"
                >
                  <div
                    class="bulma-columns"
                  >
                    <div
                      class="bulma-column"
                    >
                      <div
                        data-aos="zoom-in"
                        data-aos-delay="0"
                      >
                        <div
                          class="bulma-control  "
                        >
                          <input
                            autocomplete="fname"
                            class="bulma-input   "
                            id="firstName"
                            name="firstName"
                            placeholder="First name"
                            required=""
                            type="text"
                            value="Ester"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="bulma-column"
                    >
                      <div
                        data-aos="zoom-in"
                        data-aos-delay="0"
                      >
                        <div
                          class="bulma-control  "
                        >
                          <input
                            autocomplete="lname"
                            class="bulma-input   "
                            id="lastName"
                            name="lastName"
                            placeholder="Last name"
                            required=""
                            type="text"
                            value="Exposito"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="bulma-columns"
                  >
                    <div
                      class="bulma-column"
                    >
                      <div
                        data-aos="zoom-in"
                        data-aos-delay="0"
                      >
                        <div
                          class="bulma-control  "
                        >
                          <input
                            autocomplete="email"
                            class="bulma-input   "
                            id="email"
                            name="email"
                            placeholder="Email address"
                            required=""
                            type="text"
                            value="ester.exposito@gmail.com"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="bulma-columns"
                  >
                    <div
                      class="bulma-column"
                    >
                      <div
                        data-aos="zoom-in"
                        data-aos-delay="0"
                      >
                        <div
                          class="bulma-control  "
                        >
                          <input
                            autocomplete="subject"
                            class="bulma-input   "
                            id="subject"
                            name="subject"
                            placeholder="Subject"
                            required=""
                            type="text"
                            value="Test subject"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="bulma-columns"
                  >
                    <div
                      class="bulma-column"
                    >
                      <div
                        data-aos="zoom-in"
                        data-aos-delay="0"
                      >
                        <div
                          class="bulma-field"
                        >
                          <div
                            class="bulma-control  "
                          >
                            <textarea
                              class="bulma-textarea   bulma-has-fixed-size"
                              id="message"
                              name="message"
                              placeholder="Message"
                              required=""
                            >
                              Test message...
                            </textarea>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="bulma-content"
                  >
                    <div
                      class="bulma-notification"
                    >
                      <p
                        class="is-size-6"
                      >
                        I agree!
                      </p>
                    </div>
                    <div
                      data-aos="fade-up"
                      data-aos-delay="0"
                    >
                      <button
                        class="bulma-button bulma-is-link bulma-is-light bulma-is-fullwidth"
                        type="submit"
                      >
                        Submit
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
